WordPress/Classes/ViewRelated/Menus/Views/MenuItemSourceCell.m

```diff
@@ -0,0 +1,165 @@
+                                              [stackView.trailingAnchor constraintLessThanOrEqualToAnchor:self.contentView.trailingAnchor],
+                                              [stackView.bottomAnchor constraintEqualToAnchor:self.contentView.bottomAnchor]
+                                              ]];
+    
+    _stackView = stackView;
+}
+
+- (void)initLabelsStackView
+{
+    UIStackView *labelsStackView = [[UIStackView alloc] init];
+    labelsStackView.translatesAutoresizingMaskIntoConstraints = NO;
+    labelsStackView.distribution = UIStackViewDistributionFill;
+    labelsStackView.alignment = UIStackViewAlignmentTop;
+    labelsStackView.axis = UILayoutConstraintAxisHorizontal;
+    labelsStackView.spacing = self.stackView.spacing;
+    _labelsStackView = labelsStackView;
+
+    NSAssert(_stackView != nil, @"stackView is nil");
+    [_stackView addArrangedSubview:labelsStackView];
+}
+
+- (void)initLabel
+{
+    UILabel *label = [[UILabel alloc] init];
+    label.translatesAutoresizingMaskIntoConstraints = NO;
+    label.font = [WPStyleGuide tableviewTextFont];
+    label.textColor = [WPStyleGuide greyDarken30];
+    label.backgroundColor = [UIColor whiteColor];
+    label.numberOfLines = 0;
+    label.lineBreakMode = NSLineBreakByTruncatingTail;
+    
+    [label setContentHuggingPriority:UILayoutPriorityRequired forAxis:UILayoutConstraintAxisHorizontal];
+    [label setContentCompressionResistancePriority:UILayoutPriorityRequired forAxis:UILayoutConstraintAxisHorizontal];
+    _label = label;
+
+    NSAssert(_labelsStackView != nil, @"labelsStackView is nil");
+    [self.labelsStackView addArrangedSubview:label];
+}
+
+- (void)layoutSubviews
+{
+    [super layoutSubviews];
+    
+    [self updateSeparatorInsets];
+}
+
+- (void)setSourceSelected:(BOOL)sourceSelected
+{
+    if (_sourceSelected != sourceSelected) {
+        _sourceSelected = sourceSelected;
+        self.accessoryType = sourceSelected ? UITableViewCellAccessoryCheckmark : UITableViewCellAccessoryNone;
+    }
+}
+
+- (void)setTitle:(NSString *)title
+{
+    if (_title != title) {
+        _title = [title copy];
+        self.label.text = title;
+    }
+}
+
+- (void)setSourceHierarchyIndentation:(NSUInteger)sourceHierarchyIndentation
+{
+    if (_sourceHierarchyIndentation != sourceHierarchyIndentation) {
+        _sourceHierarchyIndentation = sourceHierarchyIndentation;
+        self.leadingLayoutConstraintForContentViewIndentation.constant = sourceHierarchyIndentation * MenuItemSourceCellHierarchyIdentationWidth;
+        [self updateSeparatorInsets];
+    }
+}
+
+- (void)updateSeparatorInsets
+{
+    CGFloat left = self.sourceHierarchyIndentation * MenuItemSourceCellHierarchyIdentationWidth;
+    left += self.stackView.layoutMargins.left;
+    self.separatorInset = UIEdgeInsetsMake(0, left, 0, 0);
+}
+
+- (CGRect)drawingRectForLabel
+{
+    CGRect rect = [self convertRect:self.label.frame fromView:self.label.superview];
```

Comment:
The idea there was if something was added to the stackView, or if the custom indentation changes for the label/content (categories), the correct rect would be returned regardless.

Currently it's a bit overkill to have the stackView here, but this has also gone through several iterations where it was handy. I'm currently thinking I'd prefer to leave it implemented for easier future iterations.